# Analysis AppStore ios
        disini data set akan di bershikan terlebih dahulu sebelum dilakukan analisis. kita bisa mencari app mana yang dengan rating terbaik. app mana dengan download terbanyak dan lainnya.
    
       tujuan dari project ini adalah membantu app developer untuk menarget pasar dan menentukan para pesaingnya di dalam appstore.


In [2]:
openedFile = open('AppleStore.csv', encoding="utf8")
from csv import reader
readFile = reader(openedFile)
appsDataApple = list(readFile)
appleHeader= appsDataApple[0]
apple = appsDataApple[1:]

In [3]:
openedFile = open('googleplaystore.csv', encoding="utf8")
from csv import reader
readFile = reader(openedFile)
appsDataAndroid = list(readFile)
androidHeader = appsDataAndroid[0]
android = appsDataAndroid[1:]

In [4]:
def exploreData(dataset, start, end, rowAndCol= False):
    dataSlice = dataset[start:end]
    for row in dataSlice:
        print(row)
        print('\n')
    if rowAndCol:
        print('number of rows',len(dataset))
        print('number of column', len(dataset[0]))

In [5]:
exploreData(appsDataAndroid,0,3, True)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


number of rows 10842
number of column 13


In [6]:
print(appsDataAndroid[10472])

['Xposed Wi-Fi-Pwd', 'PERSONALIZATION', '3.5', '1042', '404k', '100,000+', 'Free', '0', 'Everyone', 'Personalization', 'August 5, 2014', '3.0.0', '4.0.3 and up']


didalam [diskusi](https://www.kaggle.com/lava18/google-play-store-apps/discussion/66015) terjadi kesalahan rating, rating dibawah disebutkan 19 yang seharusnya rating itu dari 1-5

In [7]:
print(android[10472])

['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']


In [8]:
del android[10472]

In [9]:
print(android[10472])

['osmino Wi-Fi: free WiFi', 'TOOLS', '4.2', '134203', '4.1M', '10,000,000+', 'Free', '0', 'Everyone', 'Tools', 'August 7, 2018', '6.06.14', '4.4 and up']


# Penghapusan duplikasi


In [10]:
for app in android:
    name = app[0]
    if name == 'Instagram':
        print(app)

['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577446', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66509917', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']


terjadi duplikasi kita harus melakukan data cleaning

In [11]:
unique =[]
duplicate=[]

for app in android:
    title = app[0]
    if title in unique:
        duplicate.append(app)
    else:
        unique.append(title)
    
print('total judul duplicate' , len(duplicate))
print('contoh duplicate', duplicate[1:5])
        

total judul duplicate 1181
contoh duplicate [['Box', 'BUSINESS', '4.2', '159872', 'Varies with device', '10,000,000+', 'Free', '0', 'Everyone', 'Business', 'July 31, 2018', 'Varies with device', 'Varies with device'], ['Google My Business', 'BUSINESS', '4.4', '70991', 'Varies with device', '5,000,000+', 'Free', '0', 'Everyone', 'Business', 'July 24, 2018', '2.19.0.204537701', '4.4 and up'], ['ZOOM Cloud Meetings', 'BUSINESS', '4.4', '31614', '37M', '10,000,000+', 'Free', '0', 'Everyone', 'Business', 'July 20, 2018', '4.1.28165.0716', '4.0 and up'], ['join.me - Simple Meetings', 'BUSINESS', '4.0', '6989', 'Varies with device', '1,000,000+', 'Free', '0', 'Everyone', 'Business', 'July 16, 2018', '4.3.0.508', '4.4 and up']]


In [12]:
print(androidHeader)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


In [13]:
reviewsMax={}
for app in android:
    name = app[0]
    nReviews= float(app[3])
    if name in reviewsMax and reviewsMax[name] < nReviews:
        reviewsMax[name] = nReviews
    elif name not in reviewsMax:
        reviewsMax[name] = nReviews
print(len(reviewsMax))

9659


In [14]:
androidClean=[]
androidAdded=[]
for app in android:
    name = app[0]
    nReviews=float(app[3])
    if (nReviews == reviewsMax[name])and(name not in androidAdded):
        androidClean.append(app)
        androidAdded.append(name)

In [15]:
exploreData(androidClean,0,3,True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


number of rows 9659
number of column 13


seperti yang kita harapakan kita mendapatkan data bersih dengan rating tertinggi sebanyak 9659

In [16]:
exploreData(apple,0,3,True)

['1', '281656475', 'PAC-MAN Premium', '100788224', 'USD', '3.99', '21292', '26', '4', '4.5', '6.3.5', '4+', 'Games', '38', '5', '10', '1']


['2', '281796108', 'Evernote - stay organized', '158578688', 'USD', '0', '161065', '26', '4', '3.5', '8.2.2', '4+', 'Productivity', '37', '5', '23', '1']


['3', '281940292', 'WeatherBug - Local Weather, Radar, Maps, Alerts', '100524032', 'USD', '0', '188583', '2822', '3.5', '4.5', '5.0.0', '4+', 'Weather', '37', '5', '3', '1']


number of rows 7197
number of column 17


# Pengahapusan bahasa asing


In [43]:
def isEnglish(string):
    nonAscii=0
    for character in string:
        if ord(character) > 127:
            nonAscii +=1
    if nonAscii >3:
        return False
    else:
        return True
            
print(isEnglish('Docs To Go™ Free Office Suite'))
print(isEnglish('Instachat 😜'))

True
True


ord() adalah fungsi untuk mengetahui suatu character itu bernilai berapa.. 
contoh

In [18]:
print(ord('a'))
print(ord('爱'))


97
29233


jika melebihi batas ascii yaitu (0 - 127) maka akan mengembalikan False 

jika ingin mendeteksi bukan charcter ascii kita bisa menggunakan fungsi berikut

In [39]:
def isNonEnglish(string):
    for character in string:
        if ord(character) > 127:
            return True
    return False



In [44]:
androidEnglish =[]
appleEnglish =[]
for app in androidClean:
    name = app[0]
    if isEnglish(name):
        androidEnglish.append(app)
for app in apple:
    name = app[2]
    if isEnglish(name):
        appleEnglish.append(app)


In [45]:
exploreData(androidEnglish,0,3,True)
exploreData(appleEnglish,0,3,True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


number of rows 9614
number of column 13
['1', '281656475', 'PAC-MAN Premium', '100788224', 'USD', '3.99', '21292', '26', '4', '4.5', '6.3.5', '4+', 'Games', '38', '5', '10', '1']


['2', '281796108', 'Evernote - stay organized', '158578688', 'USD', '0', '161065', '26', '4', '3.5', '8.2.2', '4+', 'Productivity', '37', '5', '23', '1']


['3', '281940292', 'WeatherBug - Local Weather, Radar, Maps, Alerts', '100524032', 'USD', '0', '188583',